### DISCUSSION TITLE
UUIDv6 instead of time-ordered v4

### DISCUSSION THREAD

**Discussion by tillkruss** [laravel/framework#44260] [Ideas] [answered]
It would be nice to use the official [UUID Version 6](https://uuid.ramsey.dev/en/stable/rfc4122/version6.html) instead of the time-ordered v4 in an upcoming major release.

Both are sortable, but v6 makes decoding the timestamp much easier.

- Add `Str::uuid6()` helper
- Use v6 in in `HasUuids::newUniqueId()`

**Comment by juse-less** [comment, marked as answer]
UUIDv7 was merged for Laravel 10.
It's essentially the same as UUIDv6, but uses unix timestamps for the timestamp part.

#44210
